sycl/source/detail/plugin.hpp

```diff
@@ -177,17 +179,21 @@ class plugin {
     // If arguments need to be captured, then a data structure can be sent in
     // the per_instance_user_data field.
     const char *PIFnName = PiCallInfo.getFuncName();
-    uint64_t CorrelationID = pi::emitFunctionBeginTrace(PIFnName);
-    uint64_t CorrelationIDWithArgs = 0;
+    uint64_t CorrelationID = 0, CorrelationIDWithArgs = 0;
+
+    if (xptiCheckTraceEnabled(
+            PiCallStreamID,
+            (uint16_t)xpti::trace_point_type_t::function_begin)) {
+      CorrelationID = pi::emitFunctionBeginTrace(PIFnName);
```

Comment:
This will cause the issues with function_end. If no subscribers exist for function_begin CorrelationID will be 0 (invalid) and if anyone is subscribed to function_end invalid CorrelationID will be used.

Reply:
I would suggest to keep global xptiTraceEnabled here.